src/components/VTextField/VTextField.js

```diff
@@ -84,12 +84,14 @@ export default {
     },
     inputValue: {
       get () {
-        return this.value
+        return this.lazyValue
       },
       set (val) {
-        this.lazyValue = val
-        this.mask && this.setSelectionRange()
-        this.$emit('input', val)
+        // Inner unmaskText() strips of non-alphanum
+        // maskText() masks and removes dirty alphanum
+        // Outer unmaskText() provides a clean lazyValue
+        this.lazyValue = this.unmaskText(this.maskText(this.unmaskText(val)))
+        this.setSelectionRange()
```

Comment:
What about non-masked inputs? This will probably break text fields with `type="number"` as they don't support `setSelectionRange()`.

Reply:
I did look at non-mask input. I even did some tests on first with a mask then later removed it on the fly.

Yeah you're probably right about `type="number"`. Well, it's not really related to the issues this PR is trying to resolve. There are several other issues that exist even before this PR, which are overlooked yet to surface sooner or later. IMHO lets handle these separately as different issues.

Reply:
I mean `type="number"` in general. Currently `setSelectionRange` is only called if a mask is set:
```js
this.mask && this.setSelectionRange()
```

What happens with this patch if you just have a number input without a mask?

```html
<v-text-field v-model="number" type="number">
```

Reply:
Now I see what you mean.

Reply:
You can probably just put the `this.mask &&` back in.

Reply:
Placing back merely the `this.mask &&` caused the `clearable` test to fail once again. This time it's expecting an `emit()`.

Admittedly, I did not give much attention to the totally valid `null` value for input field before. So now I've fixed some more lines here and there. For the same reason, the extra check needs to be there once again @jacekkarczmarczyk : 

```javascript
if (!masked.length || !text || !text.length) return text
```